python/pyiceberg/transforms.py

```diff
@@ -418,69 +392,39 @@ class UnknownTransform(Transform):
     _source_type: IcebergType = PrivateAttr()
     _transform: str = PrivateAttr()
 
-    def __init__(self, source_type: IcebergType, transform: str, **data: Any):
+    def __init__(self, transform: str, **data: Any):
         super().__init__(**data)
-        self._source_type = source_type
         self._transform = transform
 
-    def apply(self, value: Optional[S]):
+    def hash_function(self, source: IcebergType) -> Callable[[Optional[S]], Optional[T]]:
         raise AttributeError(f"Cannot apply unsupported transform: {self}")
 
     def can_transform(self, source: IcebergType) -> bool:
-        return self._source_type == source
+        return False
```

Comment:
We may need to change this later. I'm not sure what the implication will be.

Reply:
I had the same feeling, but the `UnknownTransform` feels like a transform that you don't want to use anyway. Returning `False`, felt like the right thing to do.